test/cmd/builds.sh

```diff
@@ -12,16 +12,64 @@ os::log::install_errexit
 url=":${API_PORT:-8443}"
 project="$(oc project -q)"
 
+# expect_build_output_spec runs `oc new-build` with given arguments and expects
+# a build config with a given name to be created with a given build output spec
+function os::cmd::builds::expect_build_output_spec() {
+	if [[ $# -ne 3 ]]; then echo "os::cmd::builds::expect_build_output_spec expects three argumments, got $#"; fi
+	local newbuild_args="$1"
+  local expected_buildconfig_name="$2"
+	local expected_output_spec="$3"
+
+  local template='{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}'
+	local get_output_spec_cmd="oc get bc/${expected_buildconfig_name} --template '${template}'"
+
+	os::cmd::expect_success 'oc delete all --all'
+	os::cmd::expect_success "oc new-build ${newbuild_args}"
+	os::cmd::expect_success_and_text "${get_output_spec_cmd}" "^${expected_output_spec}\$"
+}
+
 # This test validates builds and build related commands
 
 os::cmd::expect_success 'oc new-build openshift/ruby-20-centos7 https://github.com/openshift/ruby-hello-world.git'
 os::cmd::expect_success 'oc get bc/ruby-hello-world'
 os::cmd::expect_success 'cat "${OS_ROOT}/Dockerfile" | oc new-build -D - --name=test'
 os::cmd::expect_success 'oc get bc/test'
-os::cmd::expect_success "oc new-build --dockerfile=\$'FROM centos:7\nRUN yum install -y httpd'"
-os::cmd::expect_success 'oc get bc/centos'
-os::cmd::expect_success 'oc delete all --all'
 
+# Build from Dockerfile with output to ImageStreamTag
+os::cmd::builds::expect_build_output_spec \
+	"--dockerfile=\$'FROM centos:7\nRUN yum install -y httpd'" \
```

Comment:
I kept the original test using "--dockerfile" while using "-D" for the others so that we have both short and long flag name in tests.